trusona-openid.php

```diff
@@ -13,420 +13,483 @@
+    public function login_footer()
+    {
+        if ($this->trusona_enabled) {
+            $this->disable_wp_form = apply_filters('trusona_login_footer_disable_wp_form', $this->disable_wp_form);
 
-                if ($this->disable_wp_form) {
-                    $html = ob_get_clean();
+            if ($this->disable_wp_form) {
+                $html = ob_get_clean();
 
-                    $html = $this->remove_block($html, '<p class="forgetmenot">', '</form>');
-                    $html = $this->remove_block($html, '<p id="nav">', '</p>');
-                    ob_start();
+                $html = $this->remove_block($html, '<p class="forgetmenot">', '</form>');
+                $html = $this->remove_block($html, '<p id="nav">', '</p>');
+                ob_start();
 
-                    echo $html;
-                }
+                echo $html;
             }
         }
+    }
 
-        private function remove_block($html, $selector, $end) {
-            $x = strpos($html, $selector);
-            $y = strpos($html, $end, $x);
+    private function remove_block($html, $selector, $end)
+    {
+        $x = strpos($html, $selector);
+        $y = strpos($html, $end, $x);
 
-            if ($x > 0 && $y > 0) {
-                $html = substr_replace($html, NULL, $x, ($y + strlen($end)) - $x);
-            }
+        if ($x > 0 && $y > 0) {
+            $html = substr_replace($html, null, $x, ($y + strlen($end)) - $x);
+        }
+
+        return $html;
+    }
 
-            return $html;
+    private function debug_log($message)
+    {
+        if (WP_DEBUG) {
+            error_log($message);
         }
     }
+}
 
-    new TrusonaOpenID();
+new TrusonaOpenID();
 
-    function trusona_wp_uninstall() {
-        foreach (TrusonaOpenID::$PARAMETERS as $key => $val) {
-            delete_option(TrusonaOpenID::PLUGIN_ID_PREFIX . $key);
-        }
+function trusona_wp_uninstall()
+{
+    foreach (TrusonaOpenID::$PARAMETERS as $key => $val) {
+        error_log("deleting " . TrusonaOpenID::PLUGIN_ID_PREFIX . $key);
+        delete_option(TrusonaOpenID::PLUGIN_ID_PREFIX . $key);
+    }
 
-        $users = get_users(array('meta_key' => TrusonaOpenID::PLUGIN_ID_PREFIX . 'enabled'));
+    $users = get_users(array('meta_key' => TrusonaOpenID::PLUGIN_ID_PREFIX . 'enabled'));
 
-        foreach ($users as $user) {
-            delete_user_meta($user->ID, TrusonaOpenID::PLUGIN_ID_PREFIX . 'subject_id');
-            delete_user_meta($user->ID, TrusonaOpenID::PLUGIN_ID_PREFIX . 'enabled');
-            delete_user_meta($user->ID, TrusonaOpenID::PLUGIN_ID_PREFIX . 'paired');
-        }
+    foreach ($users as $user) {
+        delete_user_meta($user->ID, TrusonaOpenID::PLUGIN_ID_PREFIX . 'subject_id');
+        delete_user_meta($user->ID, TrusonaOpenID::PLUGIN_ID_PREFIX . 'enabled');
+        delete_user_meta($user->ID, TrusonaOpenID::PLUGIN_ID_PREFIX . 'paired');
     }
-?>
```

Comment:
Is this line needed? Looks like we got rid of it

Reply:
@rtrusona that is the future of PHP
http://php.net/manual/en/language.basic-syntax.phptags.php

Reply:
But i'll bring it back just in case someone is running `5.6.0`